src/main/java/com/zotov/edu/passportofficerestservice/controller/PersonApiController.java

```diff
@@ -0,0 +1,138 @@
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/persons")
+public class PersonApiController {
+
+    private final PersonService personsService;
+
+    private final PassportService passportService;
+
+    private final PassportControllerDtoConverter passportControllerDtoConverter;
+
+    @GetMapping
+    public Page<PersonResponse> getPersons(@RequestParam(required = false) Optional<String> passportNumber,
+                                           @PageableDefault(size = 100) Pageable pageable) {
+        Page<Person> foundPersons = passportNumber
+                .map(personsService::getPersonByPassportNumber)
+                .orElseGet(() -> personsService.getAllPersons(pageable));
+        return passportControllerDtoConverter.convertPersonToDto(foundPersons);
+    }
+
+    @PostMapping()
+    @ResponseStatus(HttpStatus.CREATED)
+    public PersonResponse createPerson(@Valid @RequestBody PersonRequest person) {
+        Person createdPerson = personsService.createPerson(person.getName(),
+                person.getBirthday(), person.getCountry());
+        return passportControllerDtoConverter.convertPersonToDto(createdPerson);
+    }
+
+    @GetMapping("/{id}")
+    public PersonResponse getPerson(@PathVariable String id) {
+        return passportControllerDtoConverter.convertPersonToDto(personsService.getPerson(id));
+    }
+
+    @PutMapping("/{id}")
+    public PersonResponse updatePerson(@PathVariable String id,
+                                       @Valid @RequestBody PersonRequest person) {
+        Person updatedPerson = personsService.updatePerson(
+                id, person.getName(), person.getBirthday(), person.getCountry());
+        return passportControllerDtoConverter.convertPersonToDto(updatedPerson);
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deletePerson(@PathVariable String id) {
+        personsService.deletePersonById(id);
+    }
+
+    @GetMapping("/{id}/passports")
+    public List<PassportResponse> getPassportsByGivenDatesRange(@PathVariable String id,
+                                                                @RequestParam(required = false, defaultValue = "ACTIVE")
+                                                                        PassportState state,
+                                                                @RequestParam(required = false)
+                                                                @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate minGivenDate,
```

Comment:
It's better to declare parameter names explicitly, so that it's safe to rename variables without breaking the contract

Reply:
The name declarations have been added for the request parameters.